src/mds/SessionMap.cc

```diff
@@ -682,6 +682,9 @@ void SessionMap::touch_session(Session *session)
     by_state_entry = by_state.emplace(session->state,
 				      new xlist<Session*>).first;
   by_state_entry->second->push_back(&session->item_session_list);
+  if (session->item_custom_timeout_list.is_on_list()) {
+    custom_timeout_list.push_back(&session->item_custom_timeout_list);
+  }
```

Comment:
no need to do this

Reply:
Ah yeah I didn't finish doing it but if we try and put this in mainline I figure we'll want to maintain a map of custom timeouts->clients and then we can use that and the sorted-by-access custom_timeout_list to avoid a full traversal.